lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -212,5 +213,42 @@ def self.cleanup(method_pid, threads)
       threads.each(&:exit)
     end
     private_class_method :cleanup
+
+    def self.get_bodies_and_line_numbers(obj, aem)
+      embeds = get_embeds(obj.workspace, aem) << {:data => aem.data, :fqname => aem.fqname}
+      code_start = 0
+      line_hash = {}
+      bodies = []
+      embeds.each do |item|
+        bodies << item[:data]
+        line_count = item[:data].lines.count
+        code_end   = code_start + line_count
+        line_hash[item[:fqname]] = {:start => code_start, :end => code_end}
```

Comment:
Instead of this hash, you can use a Range object.  Then later your between can actually use the Range#include? or Range#cover? method.